### ISSUE TITLE
apache commons http library doesn't work with proxies

### ISSUE THREAD

**Issue by timo11** [specify/specify6#159]
At some point a java or apache update broke Specify's handling of proxy settings.
Any code that uses the apache commons http classes is indifferent to Specify's proxy settings.

**Comment by timo11** [comment]
see #158

**Comment by timo11** [comment]
apache commons HttpClient objects should now use Specify's proxy pref settings.

GeoLocate uses java.net http classes, don't know if it uses the proxy prefs.

Also not sure if install4j update checks works with the proxy prefs. We pass them when we call the update checker, but...

Tons of things need to be tested to verify this bug.

**Comment by timo11** [comment]
install4j uses proxy settings when the "check for updates" option on the help menu is selected, and, probably, when the automatic check for updates is done after login.

**Comment by timo11** [comment]
using a squid proxy server on my machine I have verified that stats sending, check for updates (as of commit fb93902, and the attachment server use proxy settings. And work when they use proxy settings.

The proxy settings are local prefs.
PROXY_HOST (e.g. 129.237.183.8)
PROXY_PORT (e.g. 3128)
PROXY_HOST_HTTPS
PROXY_PORT_HTTPS
PROXY_TYPE

I have only used the first two prefs. I am pretty sure the current code doesn't completely support the  others.
